### PR TITLE
RDBCL-1507 - Restoring encrypted backup times out

### PR DESCRIPTION
**Pull request by grisha-kotler**
https://issues.hibernatingrhinos.com/issue/RDBCL-1507

### CODE REVIEW

File: src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs [hunk 1]
```diff
@@ -352,6 +352,45 @@ public async Task<IOperationResult> Execute(Action<IOperationProgress> onProgres
...
+                        continue;
+
+                    restoreResult.AddError("Failed to save the database record, the restore is aborted");
+                    onProgress.Invoke(restoreResult.Progress);
+                    throw;
+                }
+            }
+
+            // we are waiting for the command to propagate to this node after it was accepted by the leader
+            await _serverStore.Cluster.WaitForIndexNotification(index, TimeSpan.FromSeconds(60));
```
Comment (karmeli87) [inline src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs]: Reaching here, doesn't ensure that the command will be committed.
This ensures that it is in the the leader's log _only_, but it has to be in the majority of the nodes in order to be committed.

Comment (karmeli87) [reply src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs]: I would move this into the `try/catch`

Comment (grisha-kotler) [reply src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs]: Maybe it should be in its own `try/catch`?
What if `WriteDatabaseRecordAsync` succeeds and we only fail in `WaitForIndexNotification`?

File: src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs [hunk 2]
```diff
@@ -295,15 +293,7 @@ public async Task<IOperationResult> Execute(Action<IOperationProgress> onProgres
 
                     // after the db for restore is done, we can safely set the db state to normal and write the DatabaseRecord
                     databaseRecord.DatabaseState = DatabaseStateStatus.Normal;
-                    var (updateIndex, _) = await _serverStore.WriteDatabaseRecordAsync(databaseName, databaseRecord, null, RaftIdGenerator.DontCareId, isRestore: true);
-                    await _serverStore.Cluster.WaitForIndexNotification(updateIndex);
-
-                    if (databaseRecord.Topology.RelevantFor(_serverStore.NodeTag))
```
Comment (ppekrol) [inline src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs:301]: Why this was removed?

Comment (grisha-kotler) [reply src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs:301]: It isn't needed anymore.
We don't wait for the database to be online in order to complete the restore operation.

**Pull request merged**